### PR TITLE
Added notes about case sensitivity of headers

### PR DESCRIPTION
**Pull request by evert**
Also made some unrelated formatting tweaks.

Please review @weierophinney

### CODE REVIEW

File: proposed/http-message.md [hunk 1]
```diff
@@ -70,7 +73,7 @@ Both `Psr\Http\Message\RequestInterface` and `Psr\Http\Message\ResponseInterface
 implemented directly, implementors SHOULD implement
 `Psr\Http\Message\RequestInterface` and `Psr\Http\Message\ResponseInterface`.
 
-From here forward, the namespace `Psr\Http\Message` will be omitted when
+From here forward, the namespace `Psr\Http\MessageInterface` will be omitted when
```
Comment (weierophinney) [inline proposed/http-message.md]: This is incorrect -- the namespace **is** `Psr\Http\Message`.

File: proposed/http-message.md [hunk 2]
```diff
@@ -97,6 +100,14 @@ echo $message->getHeader('foo');
 // Outputs: baz
 ```
 
+Despite that headers may be retrieved case-insenstively, the original case
+MUST be preserved by the implementation, in particular when retrieved with
+`getHeaders()`.
+
+Non-conforming HTTP applications may depend on a certain case, so it is useful
+for a user to be able to dictate the case of the HTTP headers when creating a 
+request or response.
```
Comment (weierophinney) [inline proposed/http-message.md]: I'd argue that this information should also be added to the interface definition itself -- under `withHeader()`, `withAddedHeader()`, and `getHeaders()`.

File: proposed/http-message.md (hunk 1 above)
Comment (evert) [inline proposed/http-message.md]: Misread this :S

File: proposed/http-message.md (hunk 2 above)
Comment (evert) [inline proposed/http-message.md]: Will do

**Comment by evert** [comment]
How does this look?

**Comment by weierophinney** [comment]
:+1: !

@pmjones Please merge this when merging #414.

**Comment by weierophinney** [comment]
@evert Can you also do a PR against php-fig/http-message with the interface changes, please?

**Comment by evert** [comment]
Yup

**Pull request merged**